Tests/Sources/VLCAudioTest.swift

```diff
@@ -0,0 +1,124 @@
+/*****************************************************************************
+ * VLCAudioTest.swift
+ *****************************************************************************
+ * Copyright (C) 2018 Mike JS. Choi
+ * Copyright (C) 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+import XCTest
+
+class VLCAudioTest: XCTestCase {
+    
+    let step = 6
+    let min: Int32 = 0
+    let max: Int32 = 200
+    
+    func testIsMuted() throws {
+        let player = VLCMediaPlayer()
+        let audio = try XCTAssertNotNilAndUnwrap(player.audio)
+        XCTAssertFalse(audio.isMuted)
+        
+        audio.isMuted = true
+        XCTAssertTrue(audio.isMuted)
+        
+        audio.isMuted = false
+        XCTAssertFalse(audio.isMuted)
+        
+        audio.setMute(true)
+        XCTAssertTrue(audio.isMuted)
+        
+        audio.setMute(false)
+        XCTAssertFalse(audio.isMuted)
+    }
+    
+    func testPassThrough() throws {
+        let player = VLCMediaPlayer()
+        let audio = try XCTAssertNotNilAndUnwrap(player.audio)
+        
+        XCTAssertFalse(audio.passthrough)
+        
+        audio.passthrough = true
+        XCTAssertTrue(audio.passthrough)
+        
+        audio.passthrough = false
+        XCTAssertFalse(audio.passthrough)
+    }
+    
+    func testVolumeDown() throws {
+        let player = VLCMediaPlayer()
+        let audio = try XCTAssertNotNilAndUnwrap(player.audio)
+        
+        audio.volume = Int32(9000)
+        XCTAssertEqual(audio.volume, max)
+        
+        let tests: [(repeatCount: Int, expected: Int32)] = [
+            (10,  140),
+            (5,   110),
+            (0,   110),
+            (15,  20),
+            (100, min)
+        ]
+        
+        for (repeatCount, expected) in tests {
+            (0..<repeatCount).forEach { _ in audio.volumeDown() }
+            XCTAssertEqual(audio.volume, expected)
```

Comment:
I just see this now but I think it might make sense to  split setting volume and volumedown.  Because right now you rely on  audio.volume = Int32(9000) to cap and work so that the next tests succeed. They should be independent.